src/featbin/extract-feature-segments.cc

```diff
@@ -133,10 +149,13 @@ int main(int argc, char *argv[]) {
       const Matrix<BaseFloat> &feats = feat_reader.Value(utterance);
       int32 num_samp = feats.NumRows(), // total number of samples present in wav data
           num_chan = feats.NumCols(); // total number of channels present in wav file
-
       // Convert start & end times of the segment to corresponding sample number
-      int32 start_samp = static_cast<int32>(start * samp_freq);
-      int32 end_samp = static_cast<int32>(end * samp_freq);
+      int32 start_samp = static_cast<int32>(round((start * samp_freq)));
```

Comment:
Yes, round() fixes a problem that was caused by static_cast to sometimes cause an incorrect number of frames to be extracted. 

Will look at the NumFrames() function.